### PR TITLE
[Mpm] add element transfer interface

### PR DESCRIPTION
**Pull request by manuelmessmer**
**Description**
To follow up PR #7473, this PR adds an interface to exchange elements and conditions between different mpi processes. 
A next step to have an mpi compatible MPM application ;)
 
**Changelog**
- Add mpm_mpi_utilitites (with `TransferElements` and `TransferConditions` functions)
- Clean up serializer of all elements and conditions inside the MPM application
- Add `CalculateOnIntegrationPoints` to `mpm_particle_point_load_condition`
- Add mpi tests for  `TransferElements` and `TransferConditions` (All available elements and conditions are tested)
- Add mpi test suite inside MPM Application and add test suite to CI.

### CODE REVIEW

File: applications/ParticleMechanicsApplication/custom_python/add_custom_utilities_to_python.cpp [hunk 1]
```diff
@@ -24,6 +24,10 @@
 #include "linear_solvers/linear_solver.h"
 #include "custom_utilities/mpm_search_element_utility.h"
 #include "custom_utilities/mpm_particle_generator_utility.h"
+#ifdef KRATOS_USING_MPI // mpi-parallel compilation
```
Comment (philbucher) [inline applications/ParticleMechanicsApplication/custom_python/add_custom_utilities_to_python.cpp]: @manuelmessmer can you refactor this into a different file such that we don't need to use the macro?

File: applications/ParticleMechanicsApplication/custom_python/add_custom_utilities_to_python.cpp [hunk 2]
```diff
@@ -58,6 +62,10 @@ namespace Python{
         m.def("SearchElement", SearchElementAccordingToDimension);
         m.def("GenerateMaterialPointElement", GenerateMaterialPointElementAccordingToDimension);
         m.def("GenerateMaterialPointCondition", &MPMParticleGeneratorUtility::GenerateMaterialPointCondition);
+#ifdef KRATOS_USING_MPI // mpi-parallel compilation
+        m.def("TransferElements", &MPM_MPI_Utilities::TransferElements);
+        m.def("TransferConditions", &MPM_MPI_Utilities::TransferConditions);
```
Comment (philbucher) [inline applications/ParticleMechanicsApplication/custom_python/add_custom_utilities_to_python.cpp]:
```suggestion
        m.def_static("TransferElements", &MPM_MPI_Utilities::TransferElements);
        m.def_static("TransferConditions", &MPM_MPI_Utilities::TransferConditions);
```

**Review by philbucher — COMMENTED** [review]
minor comment otherwise looks good to me

@roigcarlo this would probably even be sth for the core what do you think?

**Comment by roigcarlo** [comment]
Probably good idea to have in the mpi core as a part of a rebalancing algorithm. We tried it in the past but it never end up working.

I would not put it as a stand alone utility however, as probably we will prefer to use the global pointer interface for most of the things (mpm and dem for example are exceptions in which this is much better) but we can discuss the cases with the rest of @KratosMultiphysics/mpi).

**Review by philbucher — DISMISSED** [review]

**Comment by manuelmessmer** [comment]
@philbucher thanks for reviewing. Just added some Codacy suggestions in the two tests.

**Review by philbucher — APPROVED** [review]

**Pull request merged**